frontend/app/component/pollingstation/PollingStationFormNavigation.tsx

```diff
@@ -0,0 +1,186 @@
+            url = `${baseUrl}/numbers`;
+            break;
+        }
+      }
+
+      return url;
+    },
+    [election, pollingStationId],
+  );
+
+  const shouldBlock = React.useCallback<BlockerFunction>(
+    ({ currentLocation, nextLocation }) => {
+      if (currentLocation.pathname === nextLocation.pathname) {
+        return false;
+      }
+      if (!currentForm) {
+        return false;
+      }
+
+      //check if currentForm is before or same as current;
+
+      const reason = reasonBlocked(formState, currentForm, values);
+      if (reason !== null) {
+        if (reason === "changes" && formState.active === formState.current) {
+          setTemporaryCache({
+            key: currentForm.id,
+            data: currentForm.getValues(),
+          });
+          return false;
+        }
+        return true;
+      }
+
+      return false;
+    },
+    [formState, currentForm, setTemporaryCache, values],
+  );
+
+  const blocker = useBlocker(shouldBlock);
+
+  React.useEffect(() => {
+    const activeSection = formState.sections[formState.active];
+    const currentSection = formState.sections[formState.current];
+    if (activeSection && currentSection) {
+      if (activeSection.index > currentSection.index) {
+        const url = getUrlForFormSection(currentSection.id);
+        navigate(url);
+      }
+    }
+    _lastKnownSection.current = formState.active;
+  }, [formState, navigate, getUrlForFormSection]);
+
+  //check if the targetFormSection has changed and navigate to the correct url
+  React.useEffect(() => {
+    if (!targetFormSection) return;
+    if (targetFormSection !== _lastKnownSection.current) {
+      _lastKnownSection.current = targetFormSection;
+      const url = getUrlForFormSection(targetFormSection);
+      navigate(url);
+    }
+  }, [targetFormSection, getUrlForFormSection, navigate]);
+
+  return (
+    <>
+      {blocker.state === "blocked" && (
+        <Modal
+          onClose={() => {
+            blocker.proceed();
+          }}
+        >
+          <h2 id="modal-blocker-title">Wat wil je doen met je invoer?</h2>
+          <p>TEMP: {currentForm && reasonBlocked(formState, currentForm, values)}</p>
+          <p>
+            Ga je op een later moment verder met het invoeren van dit stembureau? Dan kan je de
+            invoer die je al hebt gedaan bewaren.
+            <br />
+            <br />
+            Twijfel je? Overleg dan met de coördinator.
+          </p>
+          <nav>
+            <Button
+              size="lg"
+              onClick={() => {
+                blocker.reset();
+                submitCurrentForm();
+              }}
+            >
+              Invoer bewaren
+            </Button>
+            <Button
+              size="lg"
+              variant="secondary"
+              onClick={() => {
+                blocker.proceed();
+              }}
+            >
+              Niet bewaren
+            </Button>
+          </nav>
+        </Modal>
```

Comment:
We also have this modal in `AbortDataEntryControl.tsx`. Can we re-use that one here?

Reply:
It shouldn't be the same modal, there is a different modal with different copy for this situation:

https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=3151-21014&t=NXsbLAd2Wx67Ma5j-4